drivers/crypto/crypto_si32.c

```diff
@@ -0,0 +1,1157 @@
+					   struct cipher_pkt *pkt)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	assert_dma_settings_common(channel_descriptor);
+
+	__ASSERT(channel_descriptor->SRCEND.U32 == (uintptr_t)&SI32_AES_0->DATAFIFO,
+		 "Source end pointer set to the DATAFIFO register.");
+	__ASSERT(channel_descriptor->CONFIG.DSTAIMD == 0b10,
+		 "The DSTAIMD field should be set to 010b for word increments.");
+	__ASSERT(channel_descriptor->CONFIG.SRCAIMD == 0b11,
+		 "The SRCAIMD field should be set to 011b for no increment.");
+}
+
+static void assert_dma_settings_channel_tx(struct SI32_DMADESC_A_Struct *channel_descriptor)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	assert_dma_settings_common(channel_descriptor);
+
+	__ASSERT(channel_descriptor->DSTEND.U32 == (uintptr_t)&SI32_AES_0->DATAFIFO,
+		 "Destination end pointer set to the DATAFIFO register.");
+	__ASSERT(channel_descriptor->CONFIG.DSTAIMD == 0b11,
+		 "The DSTAIMD field should be set to 011b for no increment.");
+	__ASSERT(channel_descriptor->CONFIG.SRCAIMD == 0b10,
+		 "The SRCAIMD field should be set to 010b for word increments.");
+}
+
+static void assert_dma_settings_channel_xor(struct SI32_DMADESC_A_Struct *channel_descriptor)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	assert_dma_settings_common(channel_descriptor);
+
+	__ASSERT(channel_descriptor->DSTEND.U32 == (uintptr_t)&SI32_AES_0->XORFIFO,
+		 "Destination end pointer set to the XORFIFO register.");
+	__ASSERT(channel_descriptor->CONFIG.DSTAIMD == 0b11,
+		 "The DSTAIMD field should be set to 011b for no increment.");
+	__ASSERT(channel_descriptor->CONFIG.SRCAIMD == 0b10,
+		 "The SRCAIMD field should be set to 010b for word increments.");
+}
+
+/* Set up and start input (TX) DMA channel */
+static int crypto_si32_dma_setup_tx(struct cipher_pkt *pkt, uint_fast8_t in_buf_offset)
+{
+	struct dma_block_config dma_block_cfg = {0};
+	const struct device *dma = DEVICE_DT_GET(DT_NODELABEL(dma));
+	int ret;
+
+	if (!pkt->in_len) {
+		LOG_WRN("Zero-sized data");
+		return 0;
+	}
+
+	if (pkt->in_len % 16) {
+		LOG_ERR("Data size must be 4-word aligned");
+		return -EINVAL;
+	}
+
+	dma_block_cfg.block_size = pkt->in_len - in_buf_offset;
+	dma_block_cfg.source_address = (uintptr_t)pkt->in_buf + in_buf_offset;
+	dma_block_cfg.source_addr_adj = 0b00; /* increment */
+	dma_block_cfg.dest_address = (uintptr_t)&SI32_AES_0->DATAFIFO;
+	dma_block_cfg.dest_addr_adj = 0b10; /* no change (no increment) */
+
+	struct dma_config dma_cfg = {
+		.channel_direction = MEMORY_TO_PERIPHERAL,
+		.source_data_size = 4, /* SiM3x1xx limitation: must match dest_data_size */
+		.dest_data_size = 4,   /* DATAFIFO must be written to in word chunks (4 bytes) */
+		.source_burst_length = AES_BLOCK_SIZE,
+		.dest_burst_length = AES_BLOCK_SIZE,
+		.block_count = 1,
+		.head_block = &dma_block_cfg,
+		.dma_callback = crypto_si32_dma_completed,
+	};
```

Comment:
I believe Zephyr Coding Style avoids declarations after statements.

Reply:
You are right. Fixed.